### PR TITLE
Fix defaults for autopilot config update

### LINKED ISSUE #10558: Autopilot: PUT to /v1/operator/autopilot/configuration sets unspecified params to zero/false
#### Overview of the Issue

If you attempt to update a single field of the autopilot configuration via a PUT to `/v1/operator/autopilot/configuration`, this results in all unspecified fields being set to zero/false rather than the default values.

#### Reproduction Steps

```
$ consul operator autopilot get-config
CleanupDeadServers = true
LastContactThreshold = 200ms
MaxTrailingLogs = 250
MinQuorum = 0
ServerStabilizationTime = 10s
RedundancyZoneTag = ""
DisableUpgradeMigration = false
UpgradeVersionTag = ""

$ curl -XPUT localhost:8500/v1/operator/autopilot/configuration -d '{"DisableUpgradeMigration":true}'
true

$ consul operator autopilot get-config
CleanupDeadServers = false
LastContactThreshold = 0s
MaxTrailingLogs = 0
MinQuorum = 0
ServerStabilizationTime = 0s
RedundancyZoneTag = ""
DisableUpgradeMigration = true
UpgradeVersionTag = ""
```

### Consul info for both Client and Server

This has been tested in fresh 1.6.9, 1.8.10, 1.9.7, and 1.10.0 versions.

### PR DESCRIPTION
**Pull request by jkirschner-hashicorp**
Closes #10558

Previously, for a POST request to the /v1/operator/autopilot/configuration
endpoint, any fields not included in the payload were set to a zero-initialized
value rather than the documented default value.

Now, if an optional field is not included in the payload, it will be set to its
documented default value:
```
- CleanupDeadServers:      true
- LastContactThreshold:    "200ms"
- MaxTrailingLogs:         250
- MinQuorum:               0
- ServerStabilizationTime: "10s"
- RedundancyZoneTag:       ""
- DisableUpgradeMigration: false
- UpgradeVersionTag:       ""
```

### CODE REVIEW

File: agent/operator_endpoint_test.go [hunk 1]
```diff
@@ -456,6 +456,11 @@ func TestOperator_AutopilotSetConfiguration(t *testing.T) {
 	if reply.CleanupDeadServers {
 		t.Fatalf("bad: %#v", reply)
 	}
+	if reply.LastContactThreshold != 200 * time.Millisecond ||
+	   reply.MaxTrailingLogs != 250 ||
+	   reply.ServerStabilizationTime != 10 * time.Second {
+        t.Fatalf("optional parameters not set to defaults: %#v", reply)
+   }
```
[the remark this answers is not in the document]
Comment (jkirschner-hashicorp) [reply agent/operator_endpoint_test.go:463]: Agreed that comparing all fields would be a much better test

**Comment by jkirschner-hashicorp** [comment]
Will address these comments later today - thanks!

File: api/operator_autopilot.go [hunk 2]
```diff
@@ -58,6 +58,23 @@ type AutopilotConfiguration struct {
 	ModifyIndex uint64
 }
 
+// Defines default values for the AutopilotConfiguration type, consistent with
```
Comment (jkirschner-hashicorp) [inline api/operator_autopilot.go:61]: Moved function description above declaration, per review

File: agent/operator_endpoint_test.go [hunk 3]
```diff
@@ -435,7 +435,21 @@ func TestOperator_AutopilotSetConfiguration(t *testing.T) {
 	a := NewTestAgent(t, "")
 	defer a.Shutdown()
 
+	// Provide a non-default value only for CleanupDeadServers.
+	// Expect all other fields to be updated with default values
+	// (except CreateIndex and ModifyIndex).
 	body := bytes.NewBuffer([]byte(`{"CleanupDeadServers": false}`))
+	expected := structs.AutopilotConfig{
```
Comment (jkirschner-hashicorp) [inline agent/operator_endpoint_test.go:442]: Improved test by forcing comparison of full configuration object (except CreateIndex and ModifyIndex), per review

Comment (jkirschner-hashicorp) [reply agent/operator_endpoint_test.go:442]: Side note: I got a bit tripped up between `AutopilotConfig` and `AutopilotConfiguration`

**Review by dnephin — APPROVED** [review]
LGTM!

**Pull request merged**